### PR TITLE
Add timeout configuration in settings

### PR DESCRIPTION
**Pull request by chamerling**
Hello,

On some jitsi deployments, the current 10 seconds timeout allowed before displaying error is a little too small. The current PR adds a configuration field in settings to be able to change this value as shown in the screenshots below.

Capture d’écran 2020-04-23 à 17 10 58
Capture d’écran 2020-04-23 à 17 10 46

### CODE REVIEW

**Comment by chamerling** [comment]
Hello. Any chance to get this PR reviewed?
Thanks a lot

**Comment by awlx** [comment]
I personally would hide this in an expert menu. Normal users are already flashed by the options ... and adding a timeout value might confuse them even more :/.

**Comment by saghul** [comment]
Not sure I'd have this option. 10 seconds is a very conservative value. What kind of video quality can one expect from a server which takes 10 seconds to reply to a simple request fetching a config file?

**Comment by saghul** [comment]
> I personally would hide this in an expert menu.

Agreed, if we have some "advanced options" this could go in there.

**Comment by chamerling** [comment]
> which takes 10 seconds to reply to a simple request fetching a config file?

So from the code this is not what is this timeout used for. The timeout is used from the launch of the external Jitsi client (ie the iFrame with all the jitsi assets) to the conference joined event.

**Comment by chamerling** [comment]
> Agreed, if we have some "advanced options" this could go in there.

But we do not have. So what do you suggest?

**Comment by awlx** [comment]
Create an advanced menu? :)

**Comment by saghul** [comment]
Maybe we could have a collapsible "Advanced" section in the settings. We could use this for it: https://atlaskit.atlassian.com/packages/bitbucket/panel

**Comment by lukas2511** [comment]
I think this should be a server-side option. This sounds like something where in a normal situation connectivity would be hardly usable anyway, but this would be extremely useful for something like server-side connection rate limiting, especially after a prosody crash where maybe thousands of reconnects would occur at the same time. That way they'd all need some more time for connecting, but nothing would error out and the server requests can be stretched over time.

**Comment by chamerling** [comment]
> Maybe we could have a collapsible "Advanced" section in the settings. We could use this for it: https://atlaskit.atlassian.com/packages/bitbucket/panel

OK, will try to push this today

**Comment by chamerling** [comment]
@saghul  Sorry for the delay, I pushed some changes to have the advanced settings panel like this:

Capture d’écran 2020-05-13 à 13 01 33
Capture d’écran 2020-05-13 à 13 01 28

**Comment by awlx** [comment]
That looks great!

File: package.json [hunk 1]
```diff
@@ -85,6 +85,7 @@
     "@atlaskit/navigation": "33.3.8",
     "@atlaskit/onboarding": "6.1.14",
     "@atlaskit/page": "8.0.12",
+    "@atlaskit/panel": "^0.3.5",
```
Comment (saghul) [inline package.json:88]: Can you please pin the depenency to the exact version?

Comment (chamerling) [reply package.json]: Fixed sorry for this.

**Pull request merged**